dll/win32/shell32/shlfolder.cpp

```diff
@@ -493,17 +493,35 @@ _ShowPropertiesDialogThread(LPVOID lpParameter)
 /*
  * for internal use
  */
-HRESULT WINAPI
-Shell_DefaultContextMenuCallBack(IShellFolder *psf, IDataObject *pdtobj)
+HRESULT
+SHELL32_ShowPropertiesDialog(IDataObject *pdtobj)
 {
+    if (!pdtobj)
+        return E_INVALIDARG;
+
     pdtobj->AddRef();
     if (!SHCreateThread(_ShowPropertiesDialogThread, pdtobj, CTF_INSIST | CTF_COINIT, NULL))
     {
         pdtobj->Release();
-        return HRESULT_FROM_WIN32(GetLastError());
+        return HResultFromWin32(GetLastError());
     }
     else
     {
         return S_OK;
     }
 }
+
+HRESULT
+SHELL32_DefaultContextMenuCallBack(IShellFolder *psf, IDataObject *pdo, UINT msg)
+{
+    switch (msg)
+    {
+        case DFM_MERGECONTEXTMENU:
+            return S_OK; // Yes, I want verbs
+        case DFM_INVOKECOMMAND:
+            return S_FALSE; // Do it for me please
+        case DFM_GETDEFSTATICID:
+            return S_FALSE; // Supposedly "required for Windows 7 to pick a default"
```

Comment:
Is it possible it could be also "do it for me plz"?

Reply:
No can haz that cheeseburger, ok thnx bye

Reply:
The only place I can think of where DFM_GETDEFSTATICID is actually used in Windows is for items in the recycle bin because all the other menu items perform some kind of action you might not want on double-click.

Reply:
The one in `ZipFolderMenuCallback` (in our zipfldr) is hit when running on windows 10, using our zipfldr, right-clicking on a file inside the zip.

Reply:
By used, I meant that it changes the id and returns S_OK. The callback getting DFM_GETDEFSTATICID and then doing nothing is normal and expected.